components/brave_wallet/browser/simulation_service.cc

```diff
@@ -66,63 +67,72 @@ namespace {
 
 std::string GetRelativeScanPath(const std::string& chain_id,
                                 mojom::CoinType coin) {
-  if (coin == mojom::CoinType::SOL && chain_id == mojom::kSolanaMainnet) {
-    return "solana/v0/mainnet/scan";
-  } else if (coin == mojom::CoinType::SOL &&
-             chain_id == mojom::kSolanaTestnet) {
-    return "solana/v0/testnet/scan";
-  } else if (coin == mojom::CoinType::SOL && chain_id == mojom::kSolanaDevnet) {
-    return "solana/v0/devnet/scan";
-  } else if (coin == mojom::CoinType::ETH &&
-             chain_id == mojom::kMainnetChainId) {
-    return "ethereum/v0/mainnet/scan";
-  } else if (coin == mojom::CoinType::ETH &&
-             chain_id == mojom::kGoerliChainId) {
-    return "ethereum/v0/goerli/scan";
-  } else if (coin == mojom::CoinType::ETH &&
-             chain_id == mojom::kPolygonMainnetChainId) {
-    return "polygon/v0/mainnet/scan";
-  } else if (coin == mojom::CoinType::ETH &&
-             chain_id == mojom::kBinanceSmartChainMainnetChainId) {
-    return "bnb/v0/mainnet/scan";
-  } else if (coin == mojom::CoinType::ETH &&
-             chain_id == mojom::kArbitrumMainnetChainId) {
-    return "arbitrum/v0/one/scan";
-  } else if (coin == mojom::CoinType::ETH &&
-             chain_id == mojom::kBaseMainnetChainId) {
-    return "base/v0/mainnet/scan";
+  if (coin == mojom::CoinType::SOL) {
+    if (chain_id == mojom::kSolanaMainnet) {
+      return "solana/v0/mainnet/scan";
+    } else if (chain_id == mojom::kSolanaTestnet) {
+      return "solana/v0/testnet/scan";
+    } else if (chain_id == mojom::kSolanaDevnet) {
+      return "solana/v0/devnet/scan";
+    }
+    return "";
+  }
+
+  if (coin == mojom::CoinType::ETH) {
+    if (chain_id == mojom::kArbitrumMainnetChainId) {
+      return "arbitrum/v0/one/scan";
+    } else if (chain_id == mojom::kArbitrumSepoliaChainId) {
+      return "arbitrum/v0/sepolia/scan";
+    } else if (chain_id == mojom::kAvalancheFujiTestnetChainId) {
+      return "avalanche/v0/fuji/scan";
+    } else if (chain_id == mojom::kAvalancheMainnetChainId) {
+      return "avalanche/v0/mainnet/scan";
+    } else if (chain_id == mojom::kBaseMainnetChainId) {
+      return "base/v0/mainnet/scan";
+    } else if (chain_id == mojom::kBaseSepoliaTestnetChainId) {
+      return "base/v0/sepolia/scan";
+    } else if (chain_id == mojom::kBlastMainnetChainId) {
+      return "blast/v0/mainnet/scan";
```

Comment:
This is typically done with static NoDestructor flat_map like here https://github.com/brave/brave-core/blob/master/components/brave_wallet/browser/asset_ratio_service.cc#L102-L113
so two functions wrapping two maps

Reply:
current way is ok, above is just a suggestion.

Reply:
fixed: [8374f66](https://github.com/brave/brave-core/pull/23385/commits/8374f662f447a132f0cb2c7e4b2017a09f4bc212)